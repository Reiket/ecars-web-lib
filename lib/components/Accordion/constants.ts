export const ACCORDION_TEST_ID = 'accordionTestId';
