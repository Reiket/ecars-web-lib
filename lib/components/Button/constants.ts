import type {ButtonHTMLAttributes} from 'react';

export const BUTTON_SIZE = Object.freeze({
  BIG: 'big',
  SMALL: 'small',
});

export const BUTTON_COLOR = Object.freeze({
  GREEN: 'green',
  WHITE: 'white',
  GRAY: 'gray',
});


export type ButtonAttributesType = ButtonHTMLAttributes<HTMLButtonElement>['type']
export type ButtonSizeType = (typeof BUTTON_SIZE)[keyof typeof BUTTON_SIZE];
export type ButtonColorType = (typeof BUTTON_COLOR)[keyof typeof BUTTON_COLOR];

export const BUTTON_TEST_ID = 'buttonTestId';