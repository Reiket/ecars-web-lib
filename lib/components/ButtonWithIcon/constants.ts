export const LEFT_ICON_BUTTON_TEST_ID = 'leftIconButtonTestId';
export const RIGHT_ICON_BUTTON_TEST_ID = 'rightIconButtonTestId';
