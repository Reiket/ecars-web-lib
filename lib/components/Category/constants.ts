export const CATEGORIES_LIST = Object.freeze({
  GUIDES: 'guides',
  REVIEWS: 'reviews',
  NEWS: 'news',
});

export type CategoriesListType = (typeof CATEGORIES_LIST)[keyof typeof CATEGORIES_LIST];

export const CATEGORY_TEST_ID = 'categoryTestId';
