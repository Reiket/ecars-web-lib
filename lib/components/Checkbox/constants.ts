import type {ElementProps} from '@/services/types';

export interface CheckboxProps extends ElementProps {
  name?: string;
  id?: string;
  disabled?: boolean;
  onChange?: (checked: boolean) => void;
  checked?: boolean;
  hasError?: boolean;
}

export const CHECKBOX_TEST_ID = 'checkboxTestId';
export const CHECKBOX_TEST_INPUT_ID = 'checkboxTestInputId';
