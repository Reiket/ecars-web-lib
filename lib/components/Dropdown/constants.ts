import type {ReactNode} from 'react';

export interface DropdownOption {
  value: string;
  label?: ReactNode;
}

export interface DropdownProps {
  onSelect: (value: string) => void;
  category: string;
  options: DropdownOption[];
}

export const DROPDOWN_TEST_ID = 'dropdownTestId';

export const DROPDOWN_THEME = Object.freeze({
  DARK: 'dark',
  LIGHT: 'light',
});

export type DropdownThemeType = (typeof DROPDOWN_THEME)[keyof typeof DROPDOWN_THEME];
