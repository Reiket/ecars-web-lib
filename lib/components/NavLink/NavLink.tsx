import {NavLink as Link} from 'react-router';
import type {FC, ReactNode} from 'react';
import {cn} from '@/services/helpers';
import type {ElementProps} from '@/services/types';

interface Props extends ElementProps {
  to: string;
  children: ReactNode;
}

export const NavLink: FC<Props> = ({block, to, children}) => {
  const classNames = (isActive: boolean) =>
    cn(block, 'nav-link', {
      '_active': isActive,
    });
  return (
    <Link
      className={({isActive}) => classNames(isActive)}
      to={to}
    >
      {children}
    </Link>
  );
};
