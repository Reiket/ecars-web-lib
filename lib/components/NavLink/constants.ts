export const MOCK_ROUTE_LINK = '/mock';
