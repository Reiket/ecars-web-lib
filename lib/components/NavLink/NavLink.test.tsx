import {cleanup, render} from '@testing-library/react';
import {NavLink} from './NavLink';
import '@testing-library/jest-dom';
import {MemoryRouter, Route, Routes} from 'react-router';
import {MOCK_ROUTE_LINK} from '@/components/NavLink/constants';

describe('NavLink component', () => {
  afterEach(cleanup);
  test('renders correctly with given children', () => {
    const {container, getByText} = render(
      <MemoryRouter>
        <NavLink to="/test">Test Link</NavLink>
      </MemoryRouter>,
    );

    expect(getByText('Test Link')).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });

  test("applies 'link-active' class when the link is active", () => {
    const {container, getByText} = render(
      <MemoryRouter initialEntries={[MOCK_ROUTE_LINK]}>
        <Routes>
          <Route
            path={MOCK_ROUTE_LINK}
            element={<NavLink to={MOCK_ROUTE_LINK}>Active Link</NavLink>}
          />
        </Routes>
      </MemoryRouter>,
    );

    expect(getByText('Active Link')).toHaveClass('_active');
    expect(container).toMatchSnapshot();
  });

  test("does not apply 'link-active' class when the link is not active", () => {
    const testRoute = '/other';
    const {container, getByText} = render(
      <MemoryRouter initialEntries={[testRoute]}>
        <NavLink to={MOCK_ROUTE_LINK}>Test Link</NavLink>
      </MemoryRouter>,
    );

    expect(getByText('Test Link')).not.toHaveClass('_active');
    expect(container).toMatchSnapshot();
  });
});
