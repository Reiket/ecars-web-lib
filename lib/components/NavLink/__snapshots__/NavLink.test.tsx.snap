// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavLink component applies 'link-active' class when the link is active 1`] = `
<div>
  <a
    aria-current="page"
    class="nav-link _active"
    data-discover="true"
    href="/mock"
  >
    Active Link
  </a>
</div>
`;

exports[`NavLink component does not apply 'link-active' class when the link is not active 1`] = `
<div>
  <a
    class="nav-link"
    data-discover="true"
    href="/mock"
  >
    Test Link
  </a>
</div>
`;

exports[`NavLink component renders correctly with given children 1`] = `
<div>
  <a
    class="nav-link"
    data-discover="true"
    href="/test"
  >
    Test Link
  </a>
</div>
`;
